tests/test_jsonreshape.py

```diff
@@ -0,0 +1,213 @@
+import pytest
+from bddrest import response, status, when
+
+from yhttp.core import json_reshape
+
+
+def test_jsonreshape_omit_query(app, Given):
+
+    @app.route()
+    @json_reshape(omit_queryfield='qux')
+    def get(req):
+        return dict(
+            foo='bar',
+            bar='baz',
+            baz='foo'
+        )
+
+    with Given(query=dict(qux='foo')):
+        assert status == 200
+        assert response.json == dict(bar='baz', baz='foo')
+        assert response.content_type == 'application/json'
+        assert response.headers['Content-Type'] == \
+            'application/json; charset=utf-8'
+
+        when(query=dict())
+        assert status == 200
+        assert response.json == dict(foo='bar', bar='baz', baz='foo')
+
+        when(query=dict(qux=''))
+        assert status == 200
+        assert response.json == dict(foo='bar', bar='baz', baz='foo')
+
+    @app.route('/no-omit')
+    @json_reshape(omit_queryfield=None)
+    def get(req):
+        return dict(foo='bar')
+
+    with Given('/no-omit', query={'omit-fields': 'foo'}):
+        assert status == 200
+        assert response.json == dict(foo='bar')
+
+
+def test_jsonreshape_omit(app, Given):
+
+    @app.route()
+    @json_reshape(omit='foo,bar', omit_queryfield='qux')
```

Comment:
Cannot manipulate the query string inside the `@json...` . so, remove the `omit_queryfield` feature.